mars/deploy/oscar/session.py

```diff
@@ -78,19 +81,21 @@ async def init(cls,
             from .local import new_cluster
             return (await new_cluster(address, **kwargs)).session
 
-        session_api = await SessionAPI.create(address)
+        web_address = kwargs.pop('web_address', None)
+        if web_address:
+            set_web_address(web_address)
+        session_api = await (SessionWebAPI if web_address else SessionAPI).create(address)
         # create new session
         session_address = await session_api.create_session(session_id)
-        meta_api = await MetaAPI.create(session_id, session_address)
-        task_api = await TaskAPI.create(session_id, session_address)
+        meta_api = await (MetaWebAPI if web_address else MetaAPI).create(session_id, session_address)
```

Comment:
We would like to implement separate ``MetaOscarAPI`` and ``MetaWebAPI`` basing on an abstract class ``MetaAPI``. ``create`` method will construct both subclasses given different types of addresses.